src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1706,7 +1706,13 @@ export class SavedObjectsRepository {
 
     const normalizedCounterFields = counterFields.map((counterField) => {
       const fieldName = typeof counterField === 'string' ? counterField : counterField.fieldName;
-      const incrementBy = typeof counterField === 'string' ? 1 : counterField.incrementBy || 1;
+      const incrementBy =
+        typeof counterField === 'string'
+          ? 1
+          : typeof counterField.incrementBy === 'number'
+          ? counterField.incrementBy
+          : 1;
```

Comment:
This line caused an unxpected bug `counterField.incrementBy || 1;` when `incrementBy` is `0` it'll be "falsey" and return `1` instead of `0`. Updated the logic to use the actual value of  `incrementBy` if its of type `number`